### PR TITLE
Adds Dockerfile and associated files

### LINKED ISSUE #6: Implement test suite
## Is your feature request related to a problem? Please describe.
No current tests exist.

## Describe the solution you'd like
Given the complexity of the code suite, a test suite is necessary to ensure everything works correctly. Mocking certain functions will be necessary.

### LINKED ISSUE #7: Add Dockerfile
Add a dockerfile for Travis integration.

### LINKED ISSUE #38: Add example `local_settings.cfg` file to the repository
## Is your feature request related to a problem? Please describe.
The library requires a local config file, but there's no example

## Describe the solution you'd like
Add an example file to the repository.

### PR DESCRIPTION
**Pull request by helrond**
Adds a Dockerfile with all the dependencies necessary for these scripts. Also adds a requirements.txt file and updates README. fixes #7 

Restructures the code to separate tests from working code, and adds tests, plus Travis CI file. fixes #6 

Adds a sample config file, fixes #38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,23 @@
 # iiif-image-scripts
 Repository for the RAC's iiif imaging scripts (derivative and manifest creation)
 
+## Quick start
+
+The Dockerfile included in this repository will allow you to run these scripts without install dependencies locally.
```
Comment (p-galligan) [inline README.md]:
```suggestion
The Dockerfile included in this repository will allow you to run these scripts without installing dependencies locally.
```

**Review by p-galligan — COMMENTED** [review]
Just the one minor change. I think it makes to keep working in this PR for tests if you want. We can apply a couple issues to this.

**Comment by helrond** [comment]
OK great, I'm going to plug away at tests in this branch then.

**Comment by helrond** [comment]
OK @p-galligan this is ready for a closer look.

**Review by p-galligan — APPROVED** [review]
Looks good.

From my understanding, most of the current tests are testing the run method by checking the expected number of files, or the filenames. I'm guess that when the code is cleaned up a bit and reworked that we'll need to add more tests for the various methods besides run?

**Comment by helrond** [comment]
@p-galligan yes probably. Many of those functions are in flux/may be removed so I wanted to focus on tests that would give us high-level coverage and make sure we weren't breaking things by changing variable names. We'll also want to add a test for main.py. 

I think #44 is a good place to track some of this work.

**Pull request merged**